packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3129,7 +3129,12 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
 
     final bool wasSemanticsBoundary = _semantics != null && (_cachedSemanticsConfiguration?.isSemanticBoundary ?? false);
     _cachedSemanticsConfiguration = null;
-    bool isEffectiveSemanticsBoundary = _semanticsConfiguration.isSemanticBoundary && wasSemanticsBoundary;
+    // Merger may produce sibling node to be attached to the parent of this
```

Comment:
This is non-ideal if the merger does not produce a sibling merge group. The text widget will likely need to use the merger to fix https://github.com/flutter/flutter/issues/111931. This means if the text has widget span, everytime the text widget is mark semantics dirty, it will need to go up two level and marks the grandparent dirty. Not sure if this is a big deal.

If we want to optimize this, we need two APIs, one for merge up rule one for sibling merge rule. A semanticsConfiguration only need to marks grandparent dirty if it has sibling merge rule.

Reply:
the pr to fix text widget with merger is https://github.com/flutter/flutter/pull/113461

Reply:
> If we want to optimize this, we need two APIs, one for merge up rule one for sibling merge rule. A semanticsConfiguration only need to marks grandparent dirty if it has sibling merge rule.

For my own understanding: Is this a performance issue? Or is there also a correctness issue here?

Reply:
there is also a correctness issue. 

If a renderobject has a sibling merge group, the node created by the sibling merge group will be the sibling of the semantics node this renderobject creates. If this renderobject is not explicit, then the sibling group will look up to find the first explicit rendering parent and be its sibling.

Reply:
> there is also a correctness issue.

If there's also a correctness issue, that's not great. Can we disallow the situation that causes the correctness issue?

> If a renderobject has a sibling merge group, the node created by the sibling merge group will be the sibling of the semantics node this renderobject creates. If this renderobject is not explicit, then the sibling group will look up to find the first explicit rendering parent and be its sibling.

Actually, what is the correctness issue here? Isn't this exactly what we want to achieve with this? The siblingmergegroups will become a sibling of the node that the other configurations were merged into?

Reply:
oh sorry I meant something will break if it doesn't go above one more level to mark renderobject semantics dirty